### PR TITLE
update ts across packages

### PR DESCRIPTION
**Pull request by joecowton1**
## What are you changing?

- Update TS to version 5.3.3 and tslib to 2.6.2
- Fix errors that have arisen

### CODE REVIEW

**Review by oliverabrahams — APPROVED** [review]
LGTM

File: libs/@guardian/ab-react/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
+
+### Major Changes
+
+- Update TS to version 5.3.3 and tslib to 2.6.2
+
+### Patch Changes
+
+- Updated dependencies
+  - @guardian/ab-core@7.0.0
+
```
Comment (mxdvl) [inline libs/@guardian/ab-react/CHANGELOG.md:13]: Technically, updating peer dependencies is a breaking change 😉

File: tools/nx-plugins/npm-package/package.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
 	"name": "@csnx/npm-package",
+	"version": null,
```
Comment (mxdvl) [inline tools/nx-plugins/npm-package/package.json:3]: Is this intentional? If so I do not think you want a CHANGELOG at all…

**Review by mxdvl — COMMENTED** [review]
Small notes about the fact that peer dependencies changes must always result in major version bump, never a patch, but as these are rolled into major bumps the final output looks fine.

Not explicitly approving due to notes on `prettier` and `npm-package`

File: libs/@guardian/prettier/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # @guardian/prettier
 
+## 7.0.0
+
+### Major Changes
+
+- Update TS to version 5.3.3 and tslib to 2.6.2
```
Comment (joecowton1) [inline libs/@guardian/prettier/CHANGELOG.md:7]:
```suggestion
- Update TS to version 5.3.3 and tslib to 2.6.2
```

File: libs/@guardian/source-react-components/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,16 @@
 # @guardian/source-react-components
 
+## 18.0.0
+
+### Major Changes
+
+- Update TS to version 5.3.3 and tslib to 2.6.2
+
+### Patch Changes
+
```
Comment (joecowton1) [inline libs/@guardian/source-react-components/CHANGELOG.md:10]:
```suggestion
```

File: tools/nx-plugins/npm-package/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# @csnx/npm-package
+
+## null
+
+### Patch Changes
```
Comment (joecowton1) [inline tools/nx-plugins/npm-package/CHANGELOG.md:5]:
```suggestion
## Major Changes
```

File: tools/nx-plugins/npm-package/package.json (hunk 2 above)
Comment (joecowton1) [inline tools/nx-plugins/npm-package/package.json:3]:
```suggestion
```

File: libs/@guardian/ab-react/CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,16 @@
 # @guardian/ab-react
 
+## 8.0.0
+
+### Major Changes
+
+- Update TS to version 5.3.3 and tslib to 2.6.2
+
+### Patch Changes
+
```
Comment (joecowton1) [inline libs/@guardian/ab-react/CHANGELOG.md:10]:
```suggestion
```

File: libs/@guardian/core-web-vitals/CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,16 @@
 # @guardian/core-web-vitals
 
+## 6.0.0
+
+### Major Changes
+
+- Update TS to version 5.3.3 and tslib to 2.6.2
+
+### Patch Changes
+
```
Comment (joecowton1) [inline libs/@guardian/core-web-vitals/CHANGELOG.md:10]:
```suggestion
```

File: libs/@guardian/eslint-config-typescript/CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,16 @@
 # @guardian/eslint-config-typescript
 
+## 9.0.0
+
+### Major Changes
+
+- Update TS to version 5.3.3 and tslib to 2.6.2
+
+### Patch Changes
+
```
Comment (joecowton1) [inline libs/@guardian/eslint-config-typescript/CHANGELOG.md:10]:
```suggestion
```

File: libs/@guardian/eslint-plugin-source-foundations/CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,17 @@
 # @guardian/eslint-plugin-source-foundations
 
+## 17.0.0
+
+### Major Changes
+
+- Update TS to version 5.3.3 and tslib to 2.6.2
+
+### Patch Changes
+
```
Comment (joecowton1) [inline libs/@guardian/eslint-plugin-source-foundations/CHANGELOG.md:10]:
```suggestion
```

File: libs/@guardian/eslint-plugin-source-react-components/CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,17 @@
 # @guardian/eslint-plugin-source-react-components
 
+## 21.0.0
+
+### Major Changes
+
+- Update TS to version 5.3.3 and tslib to 2.6.2
+
+### Patch Changes
+
```
Comment (joecowton1) [inline libs/@guardian/eslint-plugin-source-react-components/CHANGELOG.md:10]:
```suggestion
```

File: libs/@guardian/identity-auth/CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,16 @@
 # @guardian/identity-auth
 
+## 2.0.0
+
+### Major Changes
+
+- Update TS to version 5.3.3 and tslib to 2.6.2
+
+### Patch Changes
+
```
Comment (joecowton1) [inline libs/@guardian/identity-auth/CHANGELOG.md:10]:
```suggestion
```

File: tools/nx-plugins/npm-package/CHANGELOG.md (hunk 5 above)
Comment (joecowton1) [inline tools/nx-plugins/npm-package/CHANGELOG.md:5]:
```suggestion
```

File: tools/nx-plugins/npm-package/CHANGELOG.md [hunk 12]
```diff
@@ -0,0 +1,8 @@
+# @csnx/npm-package
+
+## null
```
[the remark this answers is not in the document]
Comment (joecowton1) [reply tools/nx-plugins/npm-package/CHANGELOG.md]: oops, removed this file

File: tools/nx-plugins/npm-package/package.json (hunk 2 above)
Comment (joecowton1) [reply tools/nx-plugins/npm-package/package.json]: also removed

File: libs/@guardian/prettier/CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (joecowton1) [reply libs/@guardian/prettier/CHANGELOG.md]: Updated. If you want to establish why tslib is needed we can do that elsewhere

File: libs/@guardian/ab-react/CHANGELOG.md (hunk 1 above)
Comment (joecowton1) [reply libs/@guardian/ab-react/CHANGELOG.md]: Ah yeah this is Changesets trying to be helpful but actually not being that helpful. We can update the changesets config to make deps changes major by default though.

**Pull request merged**